framework/source/class/qx/lang/normalize/Array.js

```diff
@@ -384,49 +384,29 @@ qx.Bootstrap.define("qx.lang.normalize.Array", {
     }
   },
 
+  /**
+   * @lint environmentNonLiteralKey()
+   */
   defer : function(statics) {
-    if (!qx.core.Environment.get("ecmascript.array.indexof")) {
-      Array.prototype.indexOf = statics.indexOf;
-    }
-
-    if (!qx.core.Environment.get("ecmascript.array.lastindexof")) {
-      Array.prototype.lastIndexOf = statics.lastIndexOf;
-    }
-
-    if (!qx.core.Environment.get("ecmascript.array.foreach")) {
-      Array.prototype.forEach = statics.forEach;
-    }
-
-    if (!qx.core.Environment.get("ecmascript.array.filter")) {
-      Array.prototype.filter = statics.filter;
-    }
-
-    if (!qx.core.Environment.get("ecmascript.array.map")) {
-      Array.prototype.map = statics.map;
-    }
-
-    if (!qx.core.Environment.get("ecmascript.array.some")) {
-      Array.prototype.some = statics.some;
-    }
-
-    if (!qx.core.Environment.get("ecmascript.array.find")) {
-      Array.prototype.find = statics.find;
-    }
-
-    if (!qx.core.Environment.get("ecmascript.array.findIndex")) {
-      Array.prototype.findIndex = statics.findIndex;
-    }
-
-    if (!qx.core.Environment.get("ecmascript.array.every")) {
-      Array.prototype.every = statics.every;
-    }
-
-    if (!qx.core.Environment.get("ecmascript.array.reduce")) {
-      Array.prototype.reduce = statics.reduce;
-    }
-
-    if (!qx.core.Environment.get("ecmascript.array.reduceright")) {
-      Array.prototype.reduceRight = statics.reduceRight;
-    }
+    var install = function(key, name) {
+      if (!qx.core.Environment.get(key)) {
```

Comment:
Wouldn't it be useful to allow ```qx.core.Environment.get``` to also accept variables as a parameter?
Does anyone know why the restriction to only use string literals is there?